src/bridge/IrcHandler.ts

```diff
@@ -137,50 +137,62 @@ export class IrcHandler {
     }
 
     /**
-     * Create a new matrix PM room for an IRC user  with nick `fromUserNick` and another
+     * Create a new matrix PM room for an IRC user with nick `fromUserNick` and another
      * matrix user with user ID `toUserId`.
-     * @param {string} toUserId The user ID of the recipient.
-     * @param {string} fromUserId The user ID of the sender.
-     * @param {string} fromUserNick The nick of the sender.
-     * @param {IrcServer} server The sending IRC server.
-     * @return {Promise} which is resolved when the PM room has been created.
+     * @param req An associated request for contextual logging.
+     * @param toUserId The user ID of the recipient.
+     * @param fromUserId The user ID of the sender.
+     * @param fromUserNick The nick of the sender.
+     * @param server The sending IRC server.
+     * @return A Promise which is resolved when the PM room has been created.
      */
-    private async createPmRoom (toUserId: string, fromUserId: string, fromUserNick: string, server: IrcServer) {
-        const users: {[userId: string]: number} = { };
-        users[toUserId] = PM_POWERLEVEL_MATRIXUSER;
-        users[fromUserId] = PM_POWERLEVEL_IRCUSER;
-        const response = await this.ircBridge.getAppServiceBridge().getIntent(
-            fromUserId
-        ).createRoom({
-            createAsClient: true,
-            options: {
-                name: (fromUserNick + " (PM on " + server.domain + ")"),
-                visibility: "private",
-                // We deliberately set our own power levels below.
-                // preset: "trusted_private_chat",
-                invite: [toUserId],
-                creation_content: {
-                    "m.federate": server.shouldFederatePMs()
-                },
-                is_direct: true,
-                initial_state: [{
-                    content: {
-                        users,
-                        events: {
-                            "m.room.avatar": 10,
-                            "m.room.name": 10,
-                            "m.room.canonical_alias": 100,
-                            "m.room.history_visibility": 100,
-                            "m.room.power_levels": 100,
-                            "m.room.encryption": 100
+    private async createPmRoom (req: BridgeRequest, toUserId: string, fromUserId: string, fromUserNick: string, server: IrcServer): Promise<MatrixRoom> {
+        let remainingAttempts = 3;
```

Comment:
Should this be configurable via an env var?

Reply:
At minimum it would be good to have a const at the top, but you can do a config option for extra points. We don't tend to do env vars at the moment.